drafts/2020-02-20-Rest-client-generator.adoc

```diff
@@ -0,0 +1,107 @@
+Pet:
+  id: 1234
+  name: Rex
+  photoUrls:
+    - https://photos.com/abcd1234
+-----
+== See Generated Code
+The generator will create model objects in Java for each of the types defined in the petstore.yaml. You can use these objects to transfer data to and from the service. For example, the Java Pet object will be, in part:
+-----
+public class Pet  {
+  private Long id;
+  private String name;
+  private List<String> photoUrls = new ArrayList<String>();
+  ...
+-----
+The generator will also create a Java interface for each endpoint. You can use these methods in your Microprofile front end to access the service. For example, to create a new pet you would call this method:
+-----
+    @POST
+    @Path("/pet")
+    @Consumes({ "application/json" })
+    public void addPet(Pet pet) throws ApiException, ProcessingException;
+-----
+While the OpenAPI Generator will create the framework, you still must implement the methods that will access the service. However, this task is much easier with the design laid out and some of the code written.
+
+== From the Command Line
+
+There are two ways to get started. The first is to download the official build and invoke the generator from the command line. See https://openapi-generator.tech/docs/installation#jar[Download Instructions].
+
+After you download the jar you can use a command similar to the following. Specify a yaml file that describes the remote service, the package names you'd like to appear in your new Java files and the directory where the output should appear.
+-----
+java -jar openapi-generator-cli-4.2.3.jar \
+  generate    \
+  --generator-name java \
+  --library microprofile \
+  --input-spec petstore.yaml \
+  --api-package org.acme.ourapp.petstore.api \
+  --model-package org.acme.ourapp.petstore.model \
+  --output project-dir
+-----
+This command will generate Java objects that represent the data the microservice recognises and Java interface files that describe methods which represent the REST endpoints. You complete the work by coding the implementations.
+
+== Use Visual Studio Code
+
+Alternatively, if you are writing your code using Visual Studio Code you can download the client generator as an extension. See the marketplace page for download and install instructions. https://marketplace.visualstudio.com/items?itemName=MicroProfile-Community.mp-rest-client-generator-vscode-ext[Visual Studio Code Marketplace].
+
+After installing the extension just right click the directory where the ouput files should be placed and select Generate a MicroProfile Rest Client. Navigate to the yaml file, confirm the directory and package names and hit enter. The generator will run and provide the Java model objects and API interfaces described previously.
+
+== Conclusion
+Integrating a new microservice into your Microprofile application can be accelerated by using the OpenAPI Generator. Take advantage of this new feature by downloading the latest version or the new Visual Studio Code extension.
```

Comment:
Good summary. Can you provide links to where they download from and get the VS Code extension from?

Reply:
Do you mean to repeat the links from the preceding sections?